packages/ra-core/src/dataProvider/populateQueryCache.ts

```diff
@@ -0,0 +1,48 @@
+import type { QueryClient } from '@tanstack/react-query';
+
+export type PopulateQueryCacheOptions = {
+    data: Record<string, any[]>;
+    queryClient: QueryClient;
+    staleTime?: number;
+};
+
+/**
+ * Populate react-query's query cache with a data dictionary
+ *
+ * @example
+ * const data = {
+ *    posts: [{ id: 1, title: 'Hello, world' }, { id: 2, title: 'FooBar' }],
+ *    comments: [{ id: 1, post_id: 1, body: 'Nice post!' }],
+ * };
+ * populateQueryCache({ data, queryClient });
+ * // setQueryData(['posts', 'getOne', { id: '1' }], { id: 1, title: 'Hello, world' });
+ * // setQueryData(['posts', 'getOne', { id: '2' }], { id: 2, title: 'FooBar' });
+ * // setQueryData(['posts', 'getMany', { ids: ['1', '2'] }], [{ id: 1, title: 'Hello, world' }, { id: 2, title: 'FooBar' }]);
+ * // setQueryData(['comments', 'getOne', { id: '1' }], { id: 1, post_id: 1, body: 'Nice post!' });
+ * // setQueryData(['comments', 'getMany', { ids: ['1'] }], [{ id: 1, post_id: 1, body: 'Nice post!' });
+ */
+export const populateQueryCache = ({
+    data,
+    queryClient,
+    staleTime = 500, // ms
+}: PopulateQueryCacheOptions) => {
```

Comment:
`staleTime = 500` might be a bit too low as a default.

On my laptop (8th get intel core i7), it takes more than 500ms to process the `CommentList` data and render the Reference fields, so the call to `getMany posts {"ids":[6,9,3,1]}` is still fired (in the demo).
If I increase the `staleTime` to 1000 ms for instance, then the prefetch works and the call to `getMany` is not fired.

More generally, this setting is a bit deep in the component tree, so it might be cumbersome to customize. I'm wondering if it would be that hard to add an extra parameter to the returned `meta` allowing to customize this `staleTime` on a per-query basis. Wdyt?

Reply:
Changed to 1000 stale time by default. Making it configurable is a nice to have, and should be done after making the *main* query stale time configurable (e.g. getList).

Reply:
Isn't it already configurable through `queryOptions`?

Reply:
The main staleTime is configurable through QueryOptions, so we may use it. But is it correct to use it for  prefetched data? Not sure. 

The idea was to allow the server to set the cache preferences, and so to use a response meta to update the cache `updated_at`. To be done sometime in the future.